superset-frontend/src/dashboard/components/filterscope/FilterScope.test.tsx

```diff
@@ -0,0 +1,372 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import userEvent from '@testing-library/user-event';
+import FilterScopeSelector from './FilterScopeSelector';
+
+const ROOT_ID = 'ROOT_ID';
+const GRID = 'GRID';
+const TABS = 'TABS';
+const TAB = 'TAB';
+const FILTER_A = 'FILTER_A';
+const FILTER_B = 'FILTER_B';
+const FILTER_C = 'FILTER_C';
+const TAB_A = 'TAB_A';
+const TAB_B = 'TAB_B';
+const CHART_A = 'CHART_A';
+const CHART_B = 'CHART_B';
+const CHART_C = 'CHART_C';
+const CHART_D = 'CHART_D';
+
+const EXPAND_ALL = 'Expand all';
+const COLLAPSE_ALL = 'Collapse all';
+const CHECKED = 'checked';
+const UNCHECKED = 'unchecked';
+const INDETERMINATE = 'indeterminate';
+const ALL_FILTERS = 'All filters';
+const ALL_CHARTS = 'All charts';
+
+const defaultProps = {
```

Comment:
Don't you think it would be nice to apply a factory here?

Reply:
Done.